blink_handler.py

```diff
@@ -104,47 +103,47 @@ def get(self, path):
     # for c in components:
     #   c.wf_urls = wf_component_content.get(c.name) or []
 
-    data = {
+    template_data = {
       'subscribers': subscribers,
       'components': components[1:] # ditch generic "Blink" component
     }
-    self.render(data, template_path=os.path.join('admin/blink.html'))
+    return template_data
 
   # Remove user from component subscribers.
-  def put(self, path):
-    params = json.loads(self.request.body)
+  @common.require_edit_permission
+  def put(self):
+    params = self.request.get_json(force=True)
     self.__update_subscribers_list(False, user_id=params.get('userId'),
                                    blink_component=params.get('componentName'),
                                    primary=params.get('primary'))
-    self.response.set_status(200, message='User removed from subscribers')
-    return self.response.write(json.dumps({'done': True}))
+    return {'done': True}
 
   # Add user to component subscribers.
-  def post(self, path):
-    params = json.loads(self.request.body)
+  @common.require_edit_permission
+  def process_post_data(self):
+    params = self.request.get_json(force=True)
 
     self.__update_subscribers_list(True, user_id=params.get('userId'),
                                    blink_component=params.get('componentName'),
                                    primary=params.get('primary'))
-    self.response.set_status(200, message='User added to subscribers')
-    return self.response.write(json.dumps(params))
+    return params
 
 
-class SubscribersHandler(common.ContentHandler):
+class SubscribersHandler(common.FlaskHandler):
+
+  TEMPLATE_PATH = 'admin/subscribers.html'
```

Comment:
Curious about this style choice.  Since this is a class bound attribute, was there a reason not to add an overloadable accessor method (versus using a public attribute with CONST style nomenclature)?

Reply:
It just emphasizes that it is supposed to be a simple value with no computation or side effects and no need to do anything analogous to calling super().  The syntax is simpler than a function would be.  For comparison, for the HTTP headers stuff uses functions so that a subclass can override it to call super() and then add another header.